### PR TITLE
Editorial: Use consistent algorithm steps to consume the head of a List

### PR DESCRIPTION
**Pull request by gibson042**
cf. [AsyncGeneratorCompleteStep](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-asyncgeneratorcompletestep) and [%ForInIteratorPrototype%.next](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-%foriniteratorprototype%.next)
> 1. Let _element_ be the first element of _list_.
> 2. Remove the first element from _list_.

### CODE REVIEW

**Comment by ljharb** [comment]
Is this common enough to warrant a prose "unshift the first element of $list" macro?

**Review by michaelficarra — APPROVED** [review]

**Comment by gibson042** [comment]
> Is this common enough to warrant a prose "unshift the first element of $list" macro?

I don't think so; it looks like only four algorithm steps do this.
```sh
$ git grep -A1 -i '^ *1[.].*first element' spec.html | grep -i remove
spec.html-                  1. Remove the first element from _remaining_.
spec.html:              1. Let _kValue_ be the first element of _values_ and remove that element from _values_.
spec.html:              1. Let _kValue_ be the first element of _values_ and remove that element from _values_.
spec.html-          1. Remove the first element from _queue_.
```

**Comment by jmdyck** [comment]
[Atomics.notify](https://tc39.es/ecma262/#sec-atomics.notify) has:
```
11.a. Let _W_ be the first agent in _S_.
11.b. Remove _W_ from the front of _S_.
```

[RemoveWaiters](https://tc39.es/ecma262/#sec-removewaiters) has:
```
4.a. Let _W_ be the first waiter in _S_.
4.c. Remove _W_ from _S_.
```
(Between them there's also `4.b. Append _W_ to _L_.` but I *think* it'd be okay to move that to after an "unshift", because `_L_` is 'local' to the AO, so can't be observed by another agent.)

**Comment by gibson042** [comment]
Good points. I've updated both, and `Atomics.notify` didn't even need List manipulation at all.

**Review by bakkot — APPROVED** [review]
LGTM. See previously https://github.com/tc39/ecma262/pull/2137/. But I'd like to wait for review from @syg on the WaiterList parts before landing.

**Review by syg — APPROVED** [review]

**Pull request merged**
